From horms  Wed Mar 26 02:37:01 2008
Return-Path: <juliusrv@googlemail.com>
X-Spam-Checker-Version: SpamAssassin 3.2.1-kirsty.vergenet.net_2007012401
	(2007-05-02) on kirsty.vergenet.net
X-Spam-Level: 
X-Spam-Status: No, score=-2.6 required=5.0 tests=BAYES_00,SPF_PASS
	autolearn=ham version=3.2.1-kirsty.vergenet.net_2007012401
X-Original-To: horms@verge.net.au
Delivered-To: horms@vergenet.net
Received: from mail.au.vergenet.net [202.4.237.240]
	by tabatha.lab.ultramonkey.org with POP3 (fetchmail-6.3.8)
	for <horms@localhost> (single-drop); Wed, 26 Mar 2008 11:37:01 +0900 (JST)
Received: from wx-out-0506.google.com (wx-out-0506.google.com [66.249.82.239])
	by kirsty.vergenet.net (Postfix) with ESMTP id C4C592408E
	for <horms@verge.net.au>; Tue, 18 Mar 2008 20:31:45 +1100 (EST)
Received: by wx-out-0506.google.com with SMTP id t14so8741969wxc.29
        for <horms@verge.net.au>; Tue, 18 Mar 2008 02:31:43 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=googlemail.com; s=gamma;
        h=domainkey-signature:received:received:message-id:date:from:to:subject:mime-version:content-type:content-transfer-encoding:content-disposition;
        bh=zKZkkf28M+cKoOwbFJhaOnvPkymMl1G+QAOBnXdcHIc=;
        b=s1iSzxAdgUQbKzvI1Hjq7aZP2m8MRTVTjfh7tlURoJUqS5Y1iJnKhT8p7iye0j6/ccbjZvBJbkAoRSuvo7UEYoR5e2SJdiHf8QjJ1vJ1K0fGqn6JNBc4qzD8Jo4xG5ELJJ+CMdG8oYr3MU/jND4RTrkxuWvxbNDcikKwywags4k=
DomainKey-Signature: a=rsa-sha1; c=nofws;
        d=googlemail.com; s=gamma;
        h=message-id:date:from:to:subject:mime-version:content-type:content-transfer-encoding:content-disposition;
        b=P4auvvQ91MMHfUp5uHajwryiKpvRdrWP+dGkuG8mXm6veK5/FNRQJ1pkuramyD0gMj8aSWv0C+UVRg4cKHa6liN+BZHBLVv1nwu+AKmpSLm96IjL3uQdFIFLcZwhaAPgwocmPA0/veKa0Hx7qfxSsHZUIP5cLuItDIl/bvFpVm0=
Received: by 10.114.80.4 with SMTP id d4mr1166627wab.44.1205832703086;
        Tue, 18 Mar 2008 02:31:43 -0700 (PDT)
Received: by 10.115.73.18 with HTTP; Tue, 18 Mar 2008 02:31:43 -0700 (PDT)
Message-ID: <441703990803180231x1fc8db5buaf836ee141fe3d53@mail.gmail.com>
Date: Tue, 18 Mar 2008 10:31:43 +0100
From: "Julius Volz" <juliusrv@googlemail.com>
To: horms@verge.net.au
Subject: [PATCH] ldirectord: Correctly declare $rows variable in check_sql()
MIME-Version: 1.0
Content-Type: text/plain; charset=ISO-8859-1
Content-Transfer-Encoding: 7bit
Content-Disposition: inline
X-Loop: oathstix_finish by tabatha.lab.ultramonkey.org

Hi Horms,

Are you the right person to submit patches for ldirectord now? Or
would I better just send them to the linux-ha-dev list (I already sent
this patch there, but it's been hanging in the moderator filter,
although I'm a list member...).

Anyways, here it goes again, in case you are the right person to talk to:

In check_sql() in ldirectord, the "$rows" variable is currently
declared incorrectly as "$row" in strict mode, leading to a Perl abort
during startup:

=========================================================
Global symbol "$rows" requires explicit package name at ldirectord line 2823.
Global symbol "$rows" requires explicit package name at ldirectord line 2831.
BEGIN not safe after errors--compilation aborted at ldirectord line 2924.

Signed-off-by: Simon Horman <horms@verge.net.au>

Fixes Debian Bug #472663
See: http://bugs.debian.org

diff -r d887e016452c ldirectord/ldirectord.in
--- a/ldirectord/ldirectord.in  Mon Mar 17 12:42:23 2008
+++ b/ldirectord/ldirectord.in  Mon Mar 17 16:48:50 2008
@@ -2766,7 +2766,7 @@
 	require DBI;
 	my ($v, $r, $dbd, $dbname) = @_;
 	my $port = ld_checkport($v, $r);
-	my ($dbh, $sth, $query, $row, $result);
+	my ($dbh, $sth, $query, $rows, $result);
 	$result = $SERVICE_DOWN;
 	$query = $$r{request};
 	$query =~ s#^/##;
